R/admiral_options.R

```diff
@@ -0,0 +1,153 @@
+admiral_options <- new.env(parent = emptyenv())
+admiral_options$subject_keys <- vars(STUDYID, USUBJID)
+# admiral_options$future_input <- vars(...) nolint
+
+######################################################
+### To enhance feature and add inputs as necessary ###
+######################################################
+
+# 1. Add additional options such as future_input as shown commented above
+# 2. Update @params with future_input in set_admiral_options roxygen documentation
+# 3. Add future_input into set_admiral_options() formals and body
+
+#' Get the Value of an Admiral Option
+#'
+#' Get the Value of an Admiral Option Which Can Be Modified for Advanced Users.
+#'
+#' @param option A character scalar of commonly used admiral function inputs.
+#'
+#'   As of now, support only available for `r enumerate(ls(admiral_options), quote_fun = dquote, conjunction = "or")`.
+#'   See `set_admiral_options()` for a description of the options.
+#'
+#' @details
+#' This function allows flexibility for function inputs that may need to be repeated
+#' multiple times in a script, such as `subject_keys`.
+#'
+#' @author Zelos Zhu
+#'
+#' @return
+#' The value of the specified option.
+#'
+#' @keywords admiral_options
+#' @family admiral_options
+#'
+#' @export
+#'
+#' @seealso [vars()], [set_admiral_options()], [derive_param_exist_flag()],
+#' [derive_param_first_event()], [derive_param_tte()], [derive_var_disposition_status()],
+#' [derive_var_dthcaus()], [derive_var_extreme_dtm()], [derive_vars_disposition_reason()],
+#' [derive_vars_period()], [create_period_dataset()]
+#'
+#'
+#' @examples
+#' library(admiral.test)
+#' library(dplyr, warn.conflicts = FALSE)
+#' data("admiral_vs")
+#' data("admiral_dm")
+#'
+#' # Merging all dm variables to vs
+#' derive_vars_merged(
+#'   admiral_vs,
+#'   dataset_add = select(admiral_dm, -DOMAIN),
+#'   by_vars = get_admiral_option("subject_keys")
+#' ) %>%
+#'   select(STUDYID, USUBJID, VSTESTCD, VISIT, VSTPT, VSSTRESN, AGE, AGEU)
+get_admiral_option <- function(option) {
+  # Check for valid option - catch function abuse
+  assert_character_scalar(option)
+
+  # Find which admiral_options is being called upon
+  possible_inputs <- ls(admiral_options)
+
+  if (option %in% possible_inputs) {
+    return(admiral_options[[option]])
+  }
+
+  # Return message otherwise, catch typos
+  err_msg <- sprintf(paste(
+    "Invalid function argument, select one of:",
+    enumerate(possible_inputs, quote_fun = dquote, conjunction = "or")
+  ))
```

Comment:
Sorry yet another one... You're not actually using `sprintf()` here. Just `paste()` should suffice.

Reply:
Np, updated!